src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -291,92 +286,103 @@ object DataModelInstanceRelation {
+          ArrayProperty(x.toVector.map(i => Float32Property(i.floatValue())))
+        case x: Iterable[java.math.BigInteger] @unchecked if x.head.isInstanceOf[java.math.BigInteger] =>
+          ArrayProperty(x.toVector.map(i => Float32Property(i.floatValue())))
+        case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, propertyType))
+      }
+      case "float64[]" | "numeric[]" => {
+        case x: Iterable[_] if x.isEmpty => ArrayProperty(Vector.empty)
+        case x: Iterable[Double] @unchecked if x.head.isInstanceOf[Double] =>
+          ArrayProperty(x.toVector.map(Float64Property))
         case x: Iterable[Float] @unchecked if x.head.isInstanceOf[Float] =>
-          Json.fromValues(x.map(value => jsonFromDouble(value.doubleValue)))
+          ArrayProperty(x.toVector.map(f => Float64Property(f.toDouble)))
+        case x: Iterable[Int] @unchecked if x.head.isInstanceOf[Int] =>
+          ArrayProperty(x.toVector.map(f => Float64Property(f.toDouble)))
         case x: Iterable[Long] @unchecked if x.head.isInstanceOf[Long] =>
-          Json.fromValues(x.map(value => jsonFromDouble(value.doubleValue)))
+          ArrayProperty(x.toVector.map(f => Float64Property(f.toDouble)))
         case x: Iterable[java.math.BigDecimal] @unchecked if x.head.isInstanceOf[java.math.BigDecimal] =>
-          Json.fromValues(x.map(value => jsonFromDouble(value.doubleValue)))
+          ArrayProperty(x.toVector.map(i => Float64Property(i.doubleValue())))
         case x: Iterable[java.math.BigInteger] @unchecked if x.head.isInstanceOf[java.math.BigInteger] =>
-          Json.fromValues(x.map(value => jsonFromDouble(value.doubleValue)))
+          ArrayProperty(x.toVector.map(i => Float64Property(i.doubleValue())))
         case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, propertyType))
       }
       case "boolean[]" => {
-        case null if !nullable =>
-          throw new CdfSparkException(propertyNotNullableMessage(propertyType)) // scalastyle:off null
-        case null => Json.Null // scalastyle:off null
-        case x: Iterable[_] if x.isEmpty => Json.arr()
+        case x: Iterable[_] if x.isEmpty => ArrayProperty(Vector.empty)
         case x: Iterable[Boolean] @unchecked if x.head.isInstanceOf[Boolean] =>
-          Json.fromValues(x.map(Json.fromBoolean))
+          ArrayProperty(x.toVector.map(BooleanProperty))
+        case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, propertyType))
+      }
+      case "int[]" | "int32[]" => {
+        case x: Iterable[_] if x.isEmpty => ArrayProperty(Vector.empty)
+        case x: Iterable[Int] @unchecked if x.head.isInstanceOf[Int] =>
+          ArrayProperty(x.toVector.map(Int32Property))
+        case x: Iterable[java.math.BigInteger] @unchecked if x.head.isInstanceOf[java.math.BigInteger] =>
+          ArrayProperty(x.toVector.map(i => Int32Property(i.intValue())))
```

Comment:
I'm not sure we should allow this. Maybe we have to because `BigInteger` is used often by Spark, even if the values are small? But it will not work well if the value is larger than `Integer.MaxValue`...

Reply:
I don't know how often they end up being inferred, so I'm a bit unsure, but I guess we can remove it for now, and add it back if we find out it's a common thing?

Reply:
I already hit class cast exception in the tests when I removed `BigDecimal` from `float32` and `BigInteger` from `int`. So I keep it if ok.